docs/reference/commands/reconfigure-node.asciidoc

```diff
@@ -0,0 +1,62 @@
+[[reconfigure-node]]
+== elasticsearch-reconfigure-node
+
+The `elasticsearch-reconfigure-node` command reconfigures an {es} node that has been installed via
+an RPM or DEB package, so that it can join an existing cluster with security features enabled.
+
+[discrete]
+=== Synopsis
+
+[source,shell]
+-----------------------------------------------------
+bin/elasticsearch-reconfigure-node
+[--enrollment-token] [-h, --help] [-E <KeyValuePair>]
+[-s, --silent] [-v, --verbose]
+-----------------------------------------------------
+
+
+[discrete]
+=== Description
+
+This command is intended for use only for DEB or RPM {es} installations
+and _before_ {es} has been started for the first time.
+
+During DEB/RPM installations, {es} assumes that the current node is the
+first node of a new cluster and configures it accordingly, enabling
+security features, generating a password for the `elastic` superuser
+and configuring TLS for the http and transport layer. However, if your
+plan is to add this node to an existing cluster that has security
+features enabled and auto-configured already, the node needs to be
+configured so that it can join the existing cluster with the help of the
+`elasticsearch-reconfigure-node` tool.
+
+You can use an enrollment token that you can generate on any of the
+nodes of your existing cluster with `elasticsearch-create-enrollment-token -s node`
+
+NOTE: You must run this tool with `sudo` so that it can edit the necessary files
+in your {es} installation configuration directory and are owned by `root:elasticsearch`.
```

Comment:
```suggestion
If you want to add a node to an existing cluster where security features are
already enabled and configured, use the `elasticsearch-reconfigure-node` tool.
You must use this tool _before_ starting {es} for the first time, or {es} will
assume that the node is a single-node cluster and enable and configure security
features by default.

On any node in the existing cluster, use the
<<create-enrollment-token,`elasticsearch-create-enrollment-token`>> tool
with the `-s node` option to generate an enrollment token, which you include as
a parameter to the the `elasticsearch-reconfigure-node` tool.

NOTE: This tool is intended only for use on DEB or RPM distributions of {es}.

You must run this tool with `sudo` so that it can edit the necessary
files in your {es} installation configuration directory that are owned by
`root:elasticsearch`.
```

Reply:
I don't think it's necessary to describe all of the security auto-configuration that happens when starting a node. This suggestion attempts to refocus the description on what you're using the tool for, and how to get the enrollment token that's needed. Happy to iterate further ✏️

Reply:
I felt it was interesting/important to know _why_ you're running this tool too, that's why I had the introductory sentences

> You must use this tool _before_ starting {es} for the first time, or {es} will
assume that the node is a single-node cluster and enable and configure security
features by default.

This implies that the "enable and configure security features" happens when the node starts, but it has actually happened during installation. And that's what I'm trying to convey here: You install with DEB/RPM and you get a node that is configured to be a secured single node cluster _on installation_. If this is not what you want, then you need to run the tool so that the node will be able to join an existing cluster the first time it stops. Happy to ge a suggestion on how this can be stated!

Reply:
Thanks for talking this through @jkakavas -- I updated my suggested change with new text, which details what happens when you install Elasticsearch, and better explains the workflow of generating the new token and how to pass that to the new node. Let me know if this change is more on the mark 🎯